### PR TITLE
Support horizontal layout in Container

### PR DESCRIPTION
**Pull request by peterstuart**
## Testing

The container example demonstrates this pretty well, but you can fiddle with the alignment and justification in it to verify that it works correctly.

### CODE REVIEW

File: src/widget/axis_size.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+use super::container::Axis;
+use embedded_graphics::prelude::*;
+
+pub trait AxisSize<T> {
+    fn for_axis(&self, axis: Axis) -> T;
+}
```
Comment (peterstuart) [inline src/widget/axis_size.rs:6]: For both `Size` and `IntrinsicSize`, get the dimension relevant to a specific axis.

File: src/widget/container/mod.rs [hunk 2]
```diff
@@ -60,28 +64,45 @@ where
...
             .children
             .iter()
             .fold(IntrinsicSize::none(), |size, widget| {
                 let widget_size = widget.intrinsic_size();
 
-                let width = match (size.width, widget_size.width) {
-                    (Some(width), Some(widget_width)) => Some(width.max(widget_width)),
-                    (Some(width), None) => Some(width),
-                    (None, Some(widget_width)) => Some(widget_width),
+                let cross_axis_size = match (
```
Comment (peterstuart) [inline src/widget/container/mod.rs]: I don't love overloading the word "Size" to mean both a width+height and a u32, but I'm not sure what the term is that means "either width or height, whichever is relevant to some axis".

File: src/widget/container/mod.rs [hunk 3]
```diff
@@ -60,28 +64,45 @@ where
...
+                    size.for_axis(self.cross_axis()),
+                    widget_size.for_axis(self.cross_axis()),
+                ) {
+                    (Some(size), Some(widget_size)) => Some(size.max(widget_size)),
+                    (Some(size), None) => Some(size),
+                    (None, Some(widget_size)) => Some(widget_size),
                     (None, None) => None,
```
Comment (peterstuart) [inline src/widget/container/mod.rs:89]: This is pretty much unchanged other than rewording it to make sense for either dimension.

File: src/widget/container/mod.rs (hunk 2 above)
Comment (helixbass) [reply src/widget/container/mod.rs]: Dimension?

**Review by helixbass — APPROVED** [review]
@peterstuart looks nice

**Pull request merged**